contracts/lib/LibPrices.sol

```diff
@@ -38,53 +38,91 @@ library Prices {
-        return int256((averageTracerPrice - averageOraclePrice) / 90);
+        return
+            (averageTracerPrice.toInt256() - averageOraclePrice.toInt256()) /
+            90;
     }
 
+    /**
+     * @notice Calculate the average price of trades in a PriceInstant instance
+     * @param price Current cumulative price and number of trades in a time period
+     * @return Average price for given instance
+     */
     function averagePrice(PriceInstant memory price)
         public
         pure
         returns (uint256)
     {
+        // todo double check safety of this.
+        // average price == 0 is not neccesarily the
+        // same as no trades in average
         if (price.trades == 0) {
             return 0;
         }
         return price.cumulativePrice / price.trades;
     }
 
+    /**
+     * @notice Calculates average price over a time period of 24 hours
+     * @dev Ignores hours where the number of trades is zero
+     * @param prices Array of PriceInstant instances in the 24 hour period
+     * @return Average price in the time period (non-weighted)
+     */
     function averagePriceForPeriod(PriceInstant[24] memory prices)
         public
         pure
         returns (uint256)
     {
-        uint256 n = prices.length <= 24 ? prices.length : 24;
         uint256[] memory averagePrices = new uint256[](24);
 
-        for (uint256 i = 0; i < n; i++) {
+        // TODO: make sure this procedure is gas-optimised
+        uint256 j = 0;
+        for (uint256 i = 0; i < 24; i++) {
             PriceInstant memory currPrice = prices[i];
-            averagePrices[i] = averagePrice(currPrice);
+
+            // don't include periods that have no trades
+            if (currPrice.trades == 0) {
+                continue;
+            } else {
+                averagePrices[j] = averagePrice(currPrice);
+                j++;
+            }
         }
 
-        return LibMath.mean(averagePrices);
+        return LibMath.meanN(averagePrices, j);
     }
 
+    /**
+     * @notice Calculate new global leverage
+     * @param _globalLeverage Current global leverage
+     * @param oldLeverage Old leverage of account
+     * @param newLeverage New leverage of account
+     * @return New global leverage, calculated from the change from
+     *        the old to the new leverage for the account
+     */
     function globalLeverage(
         uint256 _globalLeverage,
         uint256 oldLeverage,
         uint256 newLeverage
     ) public pure returns (uint256) {
-        bool leverageHasIncreased = newLeverage > oldLeverage;
-
-        if (leverageHasIncreased) {
-            return _globalLeverage + (newLeverage - oldLeverage);
-        } else {
-            return _globalLeverage - (newLeverage - oldLeverage);
+        int256 newGlobalLeverage =
+            int256(_globalLeverage) +
+                (int256(newLeverage) - int256(oldLeverage));
+
+        // note: this would require a bug in how account leverage was recorded
```

Comment:
If global leverage is a stored contract variable, then this may be eligible for an Echidna predicate.